client/state/plugins/installed/test/selectors-ts.ts

```diff
@@ -0,0 +1,193 @@
+import deepFreeze from 'deep-freeze';
+import {
+	DEACTIVATE_PLUGIN,
+	ENABLE_AUTOUPDATE_PLUGIN,
+	INSTALL_PLUGIN,
+} from 'calypso/lib/plugins/constants';
+import { userState } from 'calypso/state/selectors/test/fixtures/user-state';
+import { getAllPluginsIndexedByPluginSlug, getAllPluginsIndexedBySiteId } from '../selectors-ts';
+import { akismet, helloDolly, jetpack } from './fixtures/plugins';
+
+const siteOneId = 12345;
+const siteTwoId = 54321;
+const siteThreeId = 21435;
+
+const nonExistingSiteId = 0;
+
+const createError = function ( error, message, name = false ) {
+	const errorObj = new Error( message );
+	errorObj.name = name || error;
+	return errorObj;
+};
+
+const state = deepFreeze( {
```

Comment:
If we're using `deepFreeze` to ensure we're not introducing any unnecessary mutation, should we use it to freeze the rest of the fixture objects?

Reply:
Yeah, that's probably a good idea.